### PR TITLE
[FLINK-2362] - distinct is missing in DataSet API documentation

### PR DESCRIPTION
**Pull request by pp86**
add scala and java documentation for the distinct transformation.

### CODE REVIEW

File: docs/apis/dataset_transformations.md [hunk 1]
```diff
@@ -924,6 +924,183 @@ Not supported.
 
 **Note:** Extending the set of supported aggregation functions is on our roadmap.
 
+### Distinct
+
+The Distinct transformation computes the DataSet of the distinct elements of the source DataSet.
+The following code removes from the DataSet the duplicate elements:
```
Comment (mxm) [inline docs/apis/dataset_transformations.md]: I'd rephrase this sentence to
> The following code removes duplicate elements from the DataSet:

**Comment by mxm** [comment]
Thanks for improving the documentation. Looks good to merge for me.

**Comment by pp86** [comment]
Thanks for the suggestion @mxm , I updated the documentation.

File: docs/apis/programming_guide.md [hunk 2]
```diff
@@ -606,7 +606,17 @@ DataSet<Tuple3<Integer, String, Double>> output = input.sum(0).andMin(2);
       </td>
     </tr>
 
+    <tr>
+      <td><strong>Distinct</strong></td>
+      <td>
+        <p>Returns the distinct elements of a data set.</p>
```
Comment (mxm) [inline docs/apis/programming_guide.md]: Could you add a sentence here to explain what "distinct" means?

**Comment by StephanEwen** [comment]
Looks good, will merge this...
